### DISCUSSION TITLE
cant get to cpu temp on any instance of glances.

### DISCUSSION THREAD

**Discussion by danktankk** [nicolargo/glances#2593] [General]
I have seen many comments about this issue when looking for a solution, but still dont see one.  Sorry if I missed something but this has been an annoyance for as long as I have been using this.  Any help to get the scroll working to actually get to the bottom of the process list and see the CPU temps would be most appreciated.  Thank you

I didnt report this is a bug because this issue has been around so long that it must be by design although I cant imagine why.

**Comment by nicolargo** [comment]
Hi @danktankk 

the CPU temperature is grab thanks to the Python PsUtil lib. So you need to have an "up-to-date" version of the lib installed on your system.

Depending of your operating system, you can test it from the command line:

```
python
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import psutil
>>> psutil.__version__
'5.9.5'
>>> psutil.sensors_temperatures()
{'acpitz': [shwtemp(label='', current=27.8, high=105.0, critical=105.0), shwtemp(label='', current=29.8, high=105.0, critical=105.0)], 'coretemp': [shwtemp(label='Package id 0', current=54.0, high=100.0, critical=100.0), shwtemp(label='Core 0', current=53.0, high=100.0, critical=100.0), shwtemp(label='Core 1', current=54.0, high=100.0, critical=100.0)], 'dell_smm': [shwtemp(label='CPU', current=56.0, high=None, critical=None), shwtemp(label='Ambient', current=44.0, high=None, critical=None), shwtemp(label='SODIMM', current=41.0, high=None, critical=None)]}
```

Can you copy/paste the result of this command line on one of your non working system ?

**Comment by danktankk** [comment]
@nicolargo

```
python3 
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import psutil
>>> psutil.__version__
'5.9.6'
>>> psutil.sensors_temperatures()
{}
```
 
neither pip3 nor psutils was installed at all.  after I installed both, the above was the result.  Thanks for replying.  I would really like to get this squared away.

**Comment by danktankk** [comment]
I just want to let you know I found that the issue is not with Glances, but with my ignorance.  The issue is with the abstraction layer between VM's and the proxmox host.  I would like to see per VM temps passed through but it is not necessary since I can see the host temps. Glances for the VM's works fine minus temp polling and that is fine with me. Sorry for the hassle.  this can be closed.

**Comment by nicolargo** [comment]
No problem @danktankk !
